### DISCUSSION TITLE
HMR is slow?

### DISCUSSION THREAD

**Discussion by jamalsoueidan** [web-infra-dev/rsbuild#4150] [General]
I just started using Rsbuild because I need the module federation with HMR in the host, didn't work with vite, so I setup rsbuild because it should be faster then vite and webpack, but its NOT, very slow.

I created a simple app, nothing beside a simple App.tsx, and changing anything take more then few sec to update compared to vite which is extremlly fast, why is that? do i need to do something in the config?

![Code_0ciJ8JvaP8](https://github.com/user-attachments/assets/2a1ebea1-9f56-4eba-a14a-018c04384ece)

**Comment by chenjiahan** [comment]
Hi, can you provide an reproduction?

**Comment by jamalsoueidan** [comment]
It seems that the module federation is causing this huge delay,,,,
